src/LatteContext/Collector/VariableCollector/AssignToArrayOfTemplateVariablesCollector.php

```diff
@@ -13,12 +16,32 @@
 use PHPStan\Type\Constant\ConstantArrayType;
 use PHPStan\Type\MixedType;
 
-final class AssignToArrayOfTemplateVariablesCollector extends AbstractAssignVariableCollector
+final class AssignToArrayOfTemplateVariablesCollector implements VariableCollectorInterface
 {
-    public function collectVariables(Assign $node, Scope $scope): array
+    private NameResolver $nameResolver;
+
+    protected TemplateTypeResolver $templateTypeResolver;
+
+    public function __construct(
+        NameResolver $nameResolver,
+        TemplateTypeResolver $templateTypeResolver
+    ) {
+        $this->nameResolver = $nameResolver;
+        $this->templateTypeResolver = $templateTypeResolver;
+    }
+
+    public function isSupported(Node $node): bool
+    {
+        return $node instanceof Assign;
+    }
+
+    /**
+     * @param Assign $node
+     */
+    public function collect(Node $node, Scope $scope): ?array
     {
         if (!($node->var instanceof Array_ || $node->var instanceof List_)) {
-            return [];
+            return null;
```

Comment:
why null? we have to handle null specially, can't merge it with all other arrays

Reply:
I see, it just add some complexicity to the code - we have to check null everywhere where it is used. Also I'm not sure if it is replaced everywhere it was needed.

Reply:
It is processed only in VariableCollector when using data from sub-collectors.

Reply:
Luckily :)